### PR TITLE
provider/azurerm: Subnet properties route table association and network security group should not be computed

### PR DESCRIPTION
**Pull request by carinadigital**
Fixes https://github.com/hashicorp/terraform/issues/8227
Rework of PR  https://github.com/hashicorp/terraform/pull/8791

When you create a `subnet` then `network security group` and `route table association` are optional and computed properties. Optional is fine, but computed is wrong as it stops the following operations.
1. Changing to a new and different subnet `route table association` after the subnet has been created, then terraform will not read this new update.
2. Removing a subnet `route table association` after subnet creation (effectively defaulting to azure default subnet routing)
3. Changing to a new and different subnet `network security group` after the subnet has been created, , then terraform will not read this new update.
4. Removing a subnet `network security group` after subnet creation (effectively defaulting to azure security group rules for the subnet.)

### CODE REVIEW

**Comment by carinadigital** [comment]
Showing failure of current code when going from assigned NSG to no NSG on the subnet
```
=== RUN   TestAccAzureRMSubnet_update
--- FAIL: TestAccAzureRMSubnet_update (125.37s)
        testing.go:265: Step 1 error: Check failed: 
Check 2/3 error: azurerm_subnet.test: Attribute network_security_group_id' expected "", got "/subscriptions/[******]/resourceGroups/acctestRG-42951285905084520/providers/Microsoft.Network/networkSecurityGroups/acctestNSG42951285905084520"
```

**Comment by carinadigital** [comment]
The now leaner PR is ready for review.

Passing test with the full PR.
```
=== RUN   TestAccAzureRMSubnet_update
--- PASS: TestAccAzureRMSubnet_update (213.86s)
PASS
ok      github.com/hashicorp/terraform/builtin/providers/azurerm        213.867s
```

File: builtin/providers/azurerm/resource_arm_subnet_test.go [hunk 1]
```diff
@@ -30,6 +31,44 @@ func TestAccAzureRMSubnet_basic(t *testing.T) {
 	})
 }
 
+func TestAccAzureRMSubnet_update(t *testing.T) {
+
+	ri := acctest.RandInt()
+	preConfig := fmt.Sprintf(testAccAzureRMSubnet_basic, ri, ri, ri, ri, ri)
+	postConfig := fmt.Sprintf(testAccAzureRMSubnet_update, ri, ri, ri, ri, ri)
+	subscriptionID := os.Getenv("ARM_SUBSCRIPTION_ID")
```
Comment (carinadigital) [inline builtin/providers/azurerm/resource_arm_subnet_test.go]: Can someone suggest how to get the subscription ID from testAccProviders ? I've been unable to.

File: builtin/providers/azurerm/resource_arm_subnet_test.go [hunk 2]
```diff
@@ -30,6 +31,44 @@ func TestAccAzureRMSubnet_basic(t *testing.T) {
...
+					resource.TestCheckResourceAttr(
+						"azurerm_subnet.test", "route_table_id", rt_id),
+				),
+			},
+			resource.TestStep{
+				Config: postConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testCheckAzureRMSubnetExists("azurerm_subnet.test"),
+					resource.TestCheckResourceAttr(
+						"azurerm_subnet.test", "network_security_group_id", ""),
```
Comment (carinadigital) [inline builtin/providers/azurerm/resource_arm_subnet_test.go]: There is a potential dependency issue when going from one NSG to another NSG on the subnet. I am still investigating this, but  I consider it a different bug. Going from `assigned NSG` to `no NSG` is still a valid update.

Comment (carinadigital) [inline builtin/providers/azurerm/resource_arm_subnet_test.go]: The issue is with resourceArmSubnetRead() and setting `network_security_group_id` & `network_security_group_id` when they are empty. 
Is there a preference to fixing and adding an extra _disappears() test here or in a PR after this has been merged?

**Comment by TStraub-rms** [comment]
We are seeing something similar with the route table not associating the subnet on an update.

If we create the subnet and route table in the same `apply` the association is created, however, when doing an update (in our case we were adding some Azure Tags), the subnet is not associated to the route table.

**Comment by carinadigital** [comment]
@TStraub-rms Could you add example TF files to https://github.com/hashicorp/terraform/issues/8227 or to another issue?
I will help me cover all use cases in this PR.

**Comment by TStraub-rms** [comment]
@carinadigital I will work on getting something posted today.

**Comment by TStraub-rms** [comment]
@carinadigital Any movement on this?
I added more detail here: https://github.com/hashicorp/terraform/issues/8227

**Comment by TStraub-rms** [comment]
Bump.

**Comment by carinadigital** [comment]
@TStraub-rms I've changed role, and no longer have access to Azure.

**Comment by TStraub-rms** [comment]
@carinadigital  bummer :( best of luck to you!

**Comment by apparentlymart** [comment]
Hello @carinadigital, and thanks for working on this!

As part of the the Terraform 0.10 release earlier this year, all of the Terraform providers were moved to their own repositories in [the terraform-providers GitHub organization](https://github.com/terraform-providers), and removed from the Terraform Core repository.

Unfortunately due to the fact that new issues and pull requests are being opened constantly, it was not possible for the various provider maintainers to merge all outstanding pull requests before this split, and there is no automatic way to migrate a pull request to a new repository.

As a result, this pull request can sadly no longer be applied as-is, and so I'm going to close it.

I see that you're no longer able to work on this due to your change in role. If someone else has the time and motivation to apply the same changes to [the `azurerm` provider repository](https://github.com/terraform-providers/terraform-provider-azurerm) and open a new PR there, the maintainers of that provider should be able to review and merge it.

Thanks again for working on this, and sorry it was not able to be merged before the provider repository changes.

**Pull request closed without merging**